### ISSUE TITLE
[idea] System jobs should mount alloc directories from jobs in different group

### ISSUE THREAD

**Issue by poll0rz** [hashicorp/nomad#537]
Hi,
The main purpose for this idea is a log shipper. There's a reference about a log shipper in: https://github.com/hashicorp/nomad/blame/b00537f6c7f457e973e21d906c27f6a76539be72/website/source/docs/jobspec/environment.html.md#L49-L51
but I see it in a different way.
I think a log shipper is a system job, not a task in the same group of every job you define, right? this way, you don't have to define repetitive job files with your microservices + an additional task for the log shipper in the same group.

What do you think?

**Comment by diptanu** [comment]
@poll0rz I think this is an interesting idea. I wonder if we can run system jobs using some form of an exec driver which doesn't do any chroot so it would be able to see the host file system and hence have access to all the alloc dirs. Of course this won't work for Docker driver based system jobs.

On the topic of logging, a PR would land soon which would let you stream logs of a task in an alloc. This won't ship the logs for you though.

**Comment by poll0rz** [comment]
@diptanu yes! that exec driver without chroot sounds good to me.
that won't work for Docker but I think that's not a problem if in a future release of Nomad, the Docker driver allows mounting volumes (docker run -v /opt/hostdir:/opt/containerdir), right?

awaiting that PR you mentioned!! thanks for your help and support again. great work.

**Comment by poll0rz** [comment]
@diptanu sorry to bother you again. any news on this subject?

I'm preparing a platform migration from CoreOS (etcd+fleet) to Consul+Nomad since it is much more powerful and scales better than the other stack, but I'm having the following problem:
My docker microservices, before starting up, mount an external volume (docker run -v /opt/hostdir:/opt/containerdir) so I can write files in /opt/hostdir and the container will have them in /opt/containerdir.
I use this method to dynamically change the microservice loglevel by changing the file content. The microservice reloads it every few seconds without having to restart the container.

I'm brainstorming some workarounds but they're too rude (for example installing consul-template in every nomad hosts, and copying the file generated to every dir in `nomad_dir/allocs/*`

Please, could you help me on this topic?
Mounting docker volumes is on the Nomad roadmap? If not, any idea that could help me with this??

Thanks in advance!!! great work man.

**Comment by dadgar** [comment]
@poll0rz mounting volumes inside docker containers is on our roadmap but not something anyone is actively working on.

Would it be possible to write the log level to a key/value in consul and have your applications hit consul periodically instead of reading a file?

**Comment by poll0rz** [comment]
@dadgar thanks for the suggestion, I'll try it yes!
btw @diptanu you mentioned that you'll make a PR related to stream logs. are you working on something like this? will it support docker? https://docs.docker.com/engine/reference/logging/overview/
or it will be just an interface for the task logs the Nomad client is managing?

thank you again guys, great work.

**Comment by dadgar** [comment]
@poll0rz: The logging work is in progress. It will also work with docker!

I am going to close this as the goal can be accomplished by using the upcoming log endpoint or by using raw_exec and scanning the alloc_dirs.
